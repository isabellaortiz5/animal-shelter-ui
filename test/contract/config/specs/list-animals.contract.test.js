import { provider } from '../init-pact';
import { Matchers } from '@pact-foundation/pact';
import { AnimalController } from '../../../../controllers';

describe('Animal Service', () => {
    describe('When a request to list all animals is made', () => {
        
        beforeAll(async () => {
            await provider.setup();
            await provider.addInteraction({
                uponReceiving: 'a request to list all animals',
                state: "has animals",
                withRequest: {
                    method: 'GET',
                    path: '/animals'
                },
                willRespondWith: {
                    status: 200,
                    body: Matchers.eachLike(
                        {
                            name: Matchers.like('manchas'),
                            breed: Matchers.like("Bengali"),
                            gender: Matchers.like("Female"),
                            vaccinated: Matchers.boolean(true)
                        }
                    )
                }
            });
        });
        it('should return the correct data', async () => {
            const response = await AnimalController.list();
            expect(response.data).toMatchSnapshot();
            await provider.verify()
        });

        afterAll(() => provider.finalize());
    });
});
