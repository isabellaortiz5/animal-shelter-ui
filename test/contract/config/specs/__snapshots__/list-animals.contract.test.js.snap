// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animal Service When a request to list all animals is made should return the correct data 1`] = `
Array [
  Object {
    "breed": "Bengali",
    "gender": "Female",
    "name": "manchas",
    "vaccinated": true,
  },
]
`;
